examples/albert/run_first_peer.py

```diff
@@ -1,51 +1,158 @@
+        collaboration_args_dict.pop('upload_interval')
+
+        no_decay = ["bias", "LayerNorm.weight"]
+        optimizer_grouped_parameters = [
+            {
+                "params": [p for n, p in self.model.named_parameters() if not any(nd in n for nd in no_decay)],
+                "weight_decay": 0.01,
+            },
+            {
+                "params": [p for n, p in self.model.named_parameters() if any(nd in n for nd in no_decay)],
+                "weight_decay": 0.0,
+            },
+        ]
+
+        opt = Lamb(
+            optimizer_grouped_parameters,
+            lr=0.00176, weight_decay=0.01, clamp_value=10000.0, debias=True,
+        )
+
+        adjusted_target_batch_size = collaboration_args_dict.pop('target_batch_size') - \
+            collaboration_args_dict.pop('batch_size_lead')
+
+        collaboration_args_dict.pop('initial_peers')
+        collaboration_args_dict.pop('trainer_uuid')
+        collaboration_args_dict.pop('dht_listen_on')
+        collaboration_args_dict.pop('statistics_expiration')
+        collaboration_args_dict.pop('endpoint')
+
+        self.collaborative_optimizer = hivemind.CollaborativeOptimizer(
+            opt=opt, dht=dht, prefix=collaboration_args_dict.pop('experiment_prefix'),
+            compression_type=hivemind.utils.CompressionType.Value(collaboration_args_dict.pop('compression')),
+            throughput=collaboration_args_dict.pop('bandwidth'),
+            target_batch_size=adjusted_target_batch_size, client_mode=collaboration_args_dict.pop('client_mode'),
+            verbose=True, start=True, **collaboration_args_dict
+        )
+        self.previous_timestamp = time.time()
+
+    def is_time_to_save_state(self, cur_step):
+        if self.save_checkpoint_step_interval is None:
+            return False
+        elif cur_step - self.previous_step >= self.save_checkpoint_step_interval:
+            return True
+        else:
+            return False
+
+    def save_state(self, cur_step):
+        self.collaborative_optimizer.load_state_from_peers()
+        self.previous_step = cur_step
+
+    def is_time_to_upload(self):
+        if self.upload_model_as is None:
+            return False
+        elif time.time() - self.previous_timestamp >= self.upload_interval:
+            return True
+        else:
+            return False
+
+    def upload_checkpoint(self):
+        self.model.save_pretrained(self.upload_model_as)
+        torch.save(self.collaborative_optimizer.opt.state_dict(), f"{self.upload_model_as}/optimizer_state.pt")
+        self.previous_timestamp = time.time()
+        try:
+            subprocess.run("git add --all".split(), check=True, cwd=self.upload_model_as)
+            subprocess.run("git commit -m Updating".split(), check=True, cwd=self.upload_model_as)
+            subprocess.run("git push".split(), check=True, cwd=self.upload_model_as)
+        except subprocess.CalledProcessError:
+            logger.warning("Error while uploading model")
```

Comment:
You can [catch and report](https://stackoverflow.com/a/24850026) the exact error